packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/ClientProvider.cs

```diff
@@ -91,19 +92,36 @@ public ClientProvider(InputClient inputClient)
                 body: new AutoPropertyBody(false),
                 enclosingType: this);
 
-            _subClientInternalConstructorParams = _apiKeyAuthField != null
+            List<ParameterProvider> _subClientInternalConstructorParamsInternal = _apiKeyAuthField != null
                 ? [PipelineProperty.AsParameter, _apiKeyAuthField.AsParameter, _endpointParameter]
                 : [PipelineProperty.AsParameter, _endpointParameter];
 
+            foreach (InputParameter p in inputClient.Parameters)
+            {
+                var parameterProvider = ClientModelPlugin.Instance.TypeFactory.CreateParameter(p);
+                if (!p.IsEndpoint && parameterProvider != PipelineProperty.AsParameter && parameterProvider != _apiKeyAuthField?.AsParameter)
+                {
+                    //InputParameter? clientParam = inputClient.Parameters.FirstOrDefault(param => param.Name == "client");
```

Comment:
Can this be removed if it's not being used? :)

Reply:
yes